doc/builders/builders.chapter.md

```diff
@@ -0,0 +1,56 @@
+# The builders {#chap-builders}
+
+A builder is a function that produces derivations.
+
+:::{.warning}
+This is not to be confused with the [`builder` argument of the Nix `derivation` primitive](https://nixos.org/manual/nix/unstable/language/derivations.html), which refers to the executable that produces the build result, or [remote builder](https://nixos.org/manual/nix/stable/advanced-topics/distributed-builds.html), which refers to the remote build machines.
+:::
+
+Such a function is usually designed to follow a specific workflow, so that one can produce a package by setting a limited set of options relevant to the particular use case instead of using the `derivation` function directly.
+
+[`stdenv.mkDerivation`](#part-stdenv) is the most widely used builder, and serves as a basis for many other builders.
+In addition, it offers various options to customize parts of the builds.
+
+There is no uniform interface for builders.
+[](#chap-trivial-builders) and [fetchers](#chap-pkgs-fetchers) have various input types for convenience, but [language- or framework-specific builders](#chap-language-support) tend to follow the style of `stdenv.mkDerivation`, which accepts an attribute set or a fixed-point function taking an attribute set.
+
+## Function-based attribute recursion support in builders {#sec-build-helpers-finalattrs}
```

Comment:
This is quite a mouthful...

How about

```suggestion
## Fixed-point parameters to build helpers {#sec-build-helpers-finalattrs}
```

@roberth @infinisil

Reply:
No real opinion here, either seems fine. I'd avoid "parameters" though

Reply:
Fixed-point build helper arguments?

Reply:
"Fixed-point arguments" sounds great.

Reply:
```suggestion
## Fixed-point arguments to build helpers {#sec-build-helpers-finalattrs}
```